File: backend-app/Lostpet/Domain/Models/Pet.cs
using Domain.Models.Auth;
using Domain.Models.Enums;

namespace Domain.Models;

public class Pet : BaseEntity
{
    public string PetName { get; set; } = string.Empty;

    public PetType PetType { get; set; }

    public PetSex PetSex { get; set; }

    public string? Breed { get; set; }

    public string? Color { get; set; }

    public PetSize PetSize { get; set; }

    public PetAgeCategory PetAgeCategory { get; set; }

    public string? Description { get; set; }

    public string? PetPhotoUrl { get; set; }

    public string? ChipNumber { get; set; }

    public int? UserId { get; set; }

    public ApplicationUser? User { get; set; }

    public ICollection<Announcement> Announcements { get; set; } = new List<Announcement>();
}

File: mobile-app/types/pet.ts
import type {
  OnboardingPetAgeCategory,
  OnboardingPetSex,
  OnboardingPetSize,
  OnboardingPetType,
} from '@/types/onboarding';

export interface Pet {
  id: number;
  petName: string;
  petType: OnboardingPetType;
  petTypeLabel: string;
  petSex: OnboardingPetSex;
  petSexLabel: string;
  petSize: OnboardingPetSize;
  petSizeLabel: string;
  petAgeCategory: OnboardingPetAgeCategory;
  petAgeCategoryLabel: string;
  breed: string | null;
  chipNumber: string | null;
  description: string | null;
  petPhotoUrl: string | null;
  createdOn: string;
}

export interface CreatePetRequest {
  petName: string;
  petType: OnboardingPetType;
  petSex: OnboardingPetSex;
  petSize: OnboardingPetSize;
  petAgeCategory: OnboardingPetAgeCategory;
  breed: string;
  chipNumber?: string;
  description?: string;
  petPhotoUrl?: string;
}
